pallets/author-mapping/src/lib.rs

```diff
@@ -248,42 +225,76 @@ pub mod pallet {
 		#[pallet::weight(<T as Config>::WeightInfo::set_keys())]
 		pub fn set_keys(origin: OriginFor<T>, keys: (NimbusId, T::Keys)) -> DispatchResult {
 			let account_id = ensure_signed(origin)?;
-			let old_author_id =
-				Self::nimbus_id_of(&account_id).ok_or(Error::<T>::OldAuthorIdNotFound)?;
-			let stored_info = MappingWithDeposit::<T>::try_get(&old_author_id)
+			let (new_author_id, keys) = keys;
+			if let Some(old_author_id) = Self::nimbus_id_of(&account_id) {
+				if old_author_id != new_author_id {
+					ensure!(
+						MappingWithDeposit::<T>::get(&new_author_id).is_none(),
+						Error::<T>::AlreadyAssociated
```

Comment:
Maybe add a better comment or change the error name.
It take some time to understand here that this error is to ensure nimbus id ownership

Reply:
done in https://github.com/PureStake/moonbeam/pull/1535/commits/f310694a27119067fa64c7eb2615ad52f90ab28d

Reply:
I think the `AlreadyAssociated` was meaningful enough

Reply:
Now both `AlreadyAssociated` and `CannotOverwriteNimbusIdIfNotYours` mean the same thing no ?